service/app/controllers/textbook/TextBookController.java

```diff
@@ -0,0 +1,34 @@
+package controllers.textbook;
+
+import controllers.bulkapimanagement.BaseBulkUploadController;
+import org.sunbird.common.models.util.BulkUploadActorOperation;
+import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.request.Request;
+import play.libs.F.Promise;
+import play.mvc.Result;
+
+/**
+ *
+ * @author gauraw
+ */
+public class TextBookController extends BaseBulkUploadController {
+
+    public Promise<Result> uploadTOC(String textBookId) {
+        try {
+            Request request = createAndInitBulkRequest(BulkUploadActorOperation.TEXTBOOK_TOC_UPLOAD.getValue(), JsonKey.TEXTBOOK, false);
```

Comment:
Use handleRequest. This controller should not inherit from BaseBulkUploadController. Rather it should inherit from BaseController.